Odata-docs/webapi-8/fundamentals/property-routing.md

```diff
@@ -0,0 +1,1005 @@
+        {
+            Id = 1,
+            Name = "Customer 1",
+            ContactPhones = new List<string> { "761-116-1865" },
+            BillingAddress = new Address { Street = "Street 1A" }
+        },
+        new Customer
+        {
+            Id = 2,
+            Name = "Customer 2",
+            ContactPhones = new List<string> { "835-791-8257" },
+            BillingAddress = new PostalAddress { Street = "2A", PostalCode = "14030" }
+        },
+        new EnterpriseCustomer
+        {
+            Id = 3,
+            Name = "Customer 3",
+            ContactPhones = new List<string> { "157-575-6005" },
+            BillingAddress = new Address { Street = "Street 3A" },
+            CreditLimit = 4200,
+            RegisteredAddress = new Address { Street = "Street 3B" },
+            ShippingAddresses = new List<Address>
+            {
+                new Address { Street = "Street 3C" }
+            }
+        },
+        new EnterpriseCustomer
+        {
+            Id = 4,
+            Name = "Customer 4",
+            ContactPhones = new List<string> { "724-096-6719" },
+            BillingAddress = new Address { Street = "Street 4A" },
+            CreditLimit = 3700,
+            RegisteredAddress = new PostalAddress { Street = "Street 4B", PostalCode = "22109" },
+            ShippingAddresses = new List<Address>
+            {
+                new Address { Street = "Street 4C" }
+            }
+        }
+    };
+}
+```
+
+## Routing conventions for properties
+In this section we cover the conventions for property routing and the controller actions (endpoints) required for the request to be routed successfully.
+
+### Retrieving a property
+The route templates for this request are:
+- `GET ~/{entityset}({key})/{property}`
+- `GET ~/{entityset}/{key}/{property}`
+- `GET ~/{singleton}/{property}`
+
+The following request returns `BillingAddress` property on customer 1:
+```http
+GET http://localhost:5000/odata/Customers(1)/BillingAddress
+```
+
+For the above request to be conventionally-routed, a controller action named `GetBillingAddress` that accepts the key parameter is expected:
+```csharp
+public ActionResult GetBillingAddress([FromRoute] int key)
+{
+    var customer = customers.SingleOrDefault(d => d.Id.Equals(key));
+
+    if (customer == null)
+    {
+        return NotFound();
+    }
+
+    return Ok(customer.BillingAddress);
+}
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers(1)/BillingAddress",
+    "Street": "Street 1A"
+}
+```
+
+### Retrieving a property initialized to an instance of a derived type
+The route templates for this request are:
+- `GET ~/{entityset}({key})/{property}/{cast}`
+- `GET ~/{entityset}/{key}/{property}/{cast}`
+- `GET ~/{singleton}/{property}/{cast}`
+
+The following request returns `BillingAddress` property (_on customer 2_) initialized to an instance of `PostalAddress`:
```

Comment:
Note sure if this is better or worse, but I feel like explicitly highlighting the "crux" of this scenario might help
```suggestion
The following request returns `BillingAddress` property (_on customer 2_) whose value is an instance of `PostalAddress`, which is derived from `Address`:
```